### PR TITLE
[DO NOT MERGE] Added json settings

### PR DESCRIPTION
**Pull request by sthirugn**
This is the first of several commits to improve robottelo config.

This commit introduces
* json config - the old ini config is converted to json config
* a readme file added to explain attributes of the json config.

### CODE REVIEW

**Comment by sthirugn** [comment]
Note that this PR is to merge to newconfig branch and not master.

**Comment by sthirugn** [comment]
test:
```py
>>> import json
>>> with open('settings.json.sample') as json_data_file:
...   data = json.load(json_data_file)
... 
>>> print data
{u'foreman': {u'admin.password': u'changeme', u'admin.username': u'admin'}, u'tests': {u'provisioning': u'True', u'manifests': u'True'}, u'clients': {u'rhel7_repo': u'http://example.com/yum/repo_files/rhel7-updates.repo', u'provisioning_server': u'test', u'rhel6_repo': u'http://example.com/yum/repo_files/rhel6-updates.repo', u'image_dir': u'/opt/robottelo/images'}, u'transitions': {u'export_tar.url': u'http://example.org/sat5_export_data.tgz'}, u'robottelo': {u'window_manager_command': u'test', u'remote': u'0', u'locale': u'en_US.UTF-8', u'virtual_display': u'0', u'verbosity': u'2', u'project': u'sat', u'screenshots.base_path': u'/tmp/robottelo/screenshots/', u'smoke': u'0', u'upstream': u'1'}, u'manifest': {u'cert_url': u'http://example.org/fake_manifest.crt', u'fake_url': u'http://example.org/valid-redhat-manifest.zip', u'key_url': u'http://example.org/fake_manifest.key'}, u'ldap': {u'username': u'test', u'passwd': u'test', u'hostname': u'test', u'grpbasedn': u'test', u'basedn': u'test'}, u'performance': {u'test.virtual_machines_list': [u'127.0.0.1', u'127.0.0.1', u'127.0.0.1', u'127.0.0.1', u'127.0.0.1'], u'test.num_syncs': u'3', u'test.sync_type': u'sync', u'test.savepoint2_enabled_repos': u'', u'csv.num_buckets': u'10', u'test.savepoint1_fresh_install': u'', u'test.cdn.address': u'http://cdn.example.com/pub', u'test.foreman.perf': u'0', u'test.target_repos': u'RHEL 6 Kickstart rpms,RHEL 6 Optional Source rpms'}, u'docker': {u'external_url': u'http://localhost.2375'}, u'server': {u'ssh.username': u'root', u'scheme': u'https', u'ssh.key_private': u'/home/whoami/.ssh/id_hudson_dsa', u'hostname': u'test', u'port': u''}, u'insights': {u'insights_el6repo': u'https://www.example.com/insights-client/repo/insights-client-6.repo', u'insights_el7repo': u'https://www.example.com/insights-client/repo/insights-client-7.repo'}}

```

File: settings.json.sample [hunk 1]
```diff
@@ -0,0 +1,67 @@
+{
+    "server":{
+        "hostname":"test",
+        "scheme":"https",
+        "port":"",
+        "ssh.key_private":"/home/whoami/.ssh/id_hudson_dsa",
```
Comment (elyezer) [inline settings.json.sample]: let's stick with one way to create names, what about ssh_key, ssh_username?

File: settings.json.sample [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        "locale":"en_US.UTF-8",
+        "remote":"0",
+        "smoke":"0",
+        "upstream":"1",
+        "verbosity":"2",
+        "screenshots.base_path":"/tmp/robottelo/screenshots/",
+        "virtual_display":"0",
+        "window_manager_command":"test"
+    },
+    "tests":{
```
Comment (elyezer) [inline settings.json.sample]: As a way to say to robottelo if some feature is avaiable or not, we can check if some key is defined. If manifest is present then that feature is "enabled".

File: settings.json.sample (hunk 1 above)
Comment (sthirugn) [inline settings.json.sample]: I can change this.

File: settings.json.sample (hunk 2 above)
Comment (sthirugn) [inline settings.json.sample]: @elyezer: I am not sure if I understand your comment.

Comment (elyezer) [inline settings.json.sample]: Let's suppose:

```json
{
  ...
  "clients": {...},
}
```

and

```json
{
  ...
}
```

In the first robottelo should assume that the clients feature is enabled and the second it is not available. With that said there is no need to define a boolean telling to run or skip some tests based on the features.

File: tests/robottelo/test_settings.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
+"""Unit tests for sample json settings"""
+import json
+import os
+
+from unittest2 import TestCase
+
+JSON_INVALID_ITEMS = ['hostname ', ' scheme', 'None', '']
```
Comment (JacobCallahan) [inline tests/robottelo/test_settings.py]: why specifically these 4 items?

**Comment by JacobCallahan** [comment]
i think this is a great switch! ACK, pending comments.

File: tests/robottelo/test_settings.py (hunk 3 above)
Comment (sthirugn) [inline tests/robottelo/test_settings.py]: This is just an initial set of invalid test data that I could think of.  We can add more whenever necessary :)

**Comment by sthirugn** [comment]
@JacobCallahan thank you for the review.  But the json config factor will be on hold for now until we decide the right type of config for robottelo.  I will send more info about this today.

File: tests/robottelo/test_settings.py (hunk 3 above)
Comment (JacobCallahan) [inline tests/robottelo/test_settings.py]: ahhh ok, just wanted to make sure that we couldn't just pick some random things from the list of valid items and inject invalid characters into them.

**Comment by sthirugn** [comment]
Closed in favor of https://github.com/SatelliteQE/robottelo/pull/2807

**Pull request closed without merging**